### PR TITLE
worked with Ark 

### PR DESCRIPTION
**Pull request by SelcukAtes**
Fixed click on D3 event and display set rows to array of performance metrics for the routes the click event and remote the duration because start time is incorrect multiple test routes display on dashboard table is now correct

### CODE REVIEW

File: react/src/App.js [hunk 1]
```diff
@@ -20,35 +20,32 @@ class App extends Component {
...
       console.log('trees', tree)
       this.setState({ treeData: [tree.root] })
     })
+    socket.on('data', data => {
+      this.setState(prevState => {
+        prevState.perfData.push(data)
+      })
+    });
     if (this.state.treeData) {
       const dimensions = this.treeContainer.getBoundingClientRect();
```
Comment (ark234) [inline react/src/App.js]: Why is this here? `this.state.treeData` will always be false because the data is not there when the component renders. If this is needed, it needs to go outside `componentDidMount`, possibly inside `render()` method.

File: extress.js [hunk 2]
```diff
@@ -14,27 +14,13 @@ const Extress = {
...
-    Extress.perfData = {
-      routeDuration: null,
-      method: null,
-      route: null,
-      min: null,
-      max: null,
-      avg: null
-    };
+    Extress.durationObj = {};
+    Extress.perfData = [];
```
Comment (ark234) [inline extress.js]: `Extress.perfData` is no longer used, please remove.

File: extress.js [hunk 3]
```diff
@@ -14,27 +14,13 @@ const Extress = {
...
   startTime: null,
-  perfData: {
-    routeDuration: null,
-    method: null,
-    route: null,
-    min: null,
-    max: null,
-    avg: null
-  },
+  perfData: [],
```
Comment (ark234) [inline extress.js]: `perfData` is no longer used, please remove.

File: extress.js [hunk 4]
```diff
@@ -44,25 +30,32 @@ const Extress = {
...
+        const { min, max, avg } = util.calculatePerfData(Extress.durationObj[testId]);
+        const testData = {
+          method: req.method,
+          route: req.originalUrl,
+          min,
+          max,
+          avg,
+
+        }
+        // Extress.perfData.push(testData)
```
Comment (ark234) [inline extress.js]: Remove

File: extress.js [hunk 5]
```diff
@@ -44,25 +30,32 @@ const Extress = {
...
         axios
-          .post('http://localhost:4050/finished', Extress.perfData) //Sends just performance object
+          .post('http://localhost:4050/finished', testData) //Sends just performance object
           // .post('http://localhost:4050/finished', Extress.tree) //Sends entire tree
```
Comment (ark234) [inline extress.js]: Remove comment

File: react/src/App.js (hunk 1 above)
Comment (SelcukAtes) [reply react/src/App.js]: This was for positioning for the D3 Canvas, Currently not working. I am going to position the Tree starting point to middle of the Canvas rather than Left Corner.

File: extress.js (hunk 2 above)
Comment (SelcukAtes) [reply extress.js]: Removed.

File: extress.js (hunk 3 above)
Comment (SelcukAtes) [reply extress.js]: removed

File: extress.js (hunk 4 above)
Comment (SelcukAtes) [reply extress.js]: removed

File: extress.js (hunk 5 above)
Comment (SelcukAtes) [reply extress.js]: Removed

File: react/src/App.js (hunk 1 above)
Comment (ark234) [reply react/src/App.js]: 👍
